content/assignments/parallel-graph/skel/os_threadpool.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __SO_THREADPOOL_H__
+#define __SO_THREADPOOL_H__
+
+#include <pthread.h>
+
+typedef struct {
+    void *argument;
+    void (*task)(void *);
+} os_task_t;
+
+typedef struct _node{
+    os_task_t *task;
+    struct _node *next;
+} os_task_queue_t;
+
+typedef struct {
+    unsigned int should_stop;
+
+    unsigned int num_threads;
+    pthread_t *threads;
+
+    os_task_queue_t *tasks;
```

Comment:
Shouldn't the `os_threadpool_t` make use of the `os_queue_t` structure defined in `os_list.h`? Otherwise what is the point of this data structure if we cannot modify the header files and use this structure as our queue of tasks?

Reply:
yeah, this seems to be a slip, however, you can use it for whatever else purposes.